### PR TITLE
HomeWork

### PR DESCRIPTION
**Pull request by Daniluko**

### CODE REVIEW

File: Lesson10/src/main/java/JDBC/CRUDTempateInterface.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package JDBC;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+@FunctionalInterface
+public interface CRUDTempateInterface<A> {
```
Comment (un-install) [inline Lesson10/src/main/java/JDBC/CRUDTempateInterface.java]: you dont realy need generic here

File: Lesson10/src/main/java/JDBC/CustomerDaoImplAbstract.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Set;
+
+public class CustomerDaoImplAbstract implements CustomerDao {
+
+    @Override
+    public Set<Customer> getAllCustomer() throws SQLException {
```
Comment (un-install) [inline Lesson10/src/main/java/JDBC/CustomerDaoImplAbstract.java]: why it is here?

File: Lesson10/src/main/java/JDBC/CustomerDaoImplAbstract.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Set;
+
+public class CustomerDaoImplAbstract implements CustomerDao {
```
Comment (un-install) [inline Lesson10/src/main/java/JDBC/CustomerDaoImplAbstract.java]: why not to extend CustomerDaoImpl? In that case you would be able to use super methods and override olds.

File: Lesson10/src/main/java/JDBC/CustomerDaoImplAbstract.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+        PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM CUSTOMERS");
+        ResultSet result = preparedStatement.executeQuery();
+
+        while (result.next()) {
+            Customer customer1 = new Customer(result.getBigDecimal("CUST_NUM"), result.getString("COMPANY"),
+                    result.getBigDecimal("CUST_REP"), result.getBigDecimal("CREDIT_LIMIT"));
+            customer.add(customer1);
+        }
+        result.close();
```
Comment (un-install) [inline Lesson10/src/main/java/JDBC/CustomerDaoImplAbstract.java]: where is finaly?

File: Lesson10/src/test/java/JDBC/CustomerDaoImplAbstractTest.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        Set<Customer> customers = customerDao.getAllCustomer();
+        System.out.println(customers);
+        assertFalse(customers.isEmpty());
+    }
+
+    @Test
+    public void testFindCustomerByCustRep()throws SQLException {
+        Set<Customer> customers = customerDao.findCustomerByCustRep(BigDecimal.valueOf(101));
+        System.out.println(customers);
+        assertEquals(customers.iterator().next().getCustRep(),BigDecimal.valueOf(101));
```
Comment (un-install) [inline Lesson10/src/test/java/JDBC/CustomerDaoImplAbstractTest.java]: ?streams¿

Comment (un-install) [reply Lesson10/src/test/java/JDBC/CustomerDaoImplAbstractTest.java]: But, actualy, is good

File: Lesson10/src/main/java/JDBC/CRUDTempateInterface.java (hunk 1 above)
Comment (SergeyPasko) [reply Lesson10/src/main/java/JDBC/CRUDTempateInterface.java]: As marker

File: Lesson10/src/main/java/JDBC/CustomerDaoImplAbstract.java (hunk 3 above)
Comment (SergeyPasko) [reply Lesson10/src/main/java/JDBC/CustomerDaoImplAbstract.java]: Bad name

File: Lesson10/src/main/java/JDBC/CustomerDaoImplInterface.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Set;
+
+public class CustomerDaoImplInterface implements CustomerDao {
```
Comment (SergeyPasko) [inline Lesson10/src/main/java/JDBC/CustomerDaoImplInterface.java]: Very bad name

File: Lesson10/src/test/java/JDBC/CustomerDaoImplAbstractTest.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import java.math.BigDecimal;
+import java.sql.SQLException;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class CustomerDaoImplAbstractTest {
+    private static final Customer CUSTOMER = new Customer(BigDecimal.valueOf(100),  "KO-KO-KO",
+            BigDecimal.valueOf(12), BigDecimal.valueOf(999));
+    private CustomerDao customerDao = new CustomerDaoImpl();
```
Comment (SergeyPasko) [inline Lesson10/src/test/java/JDBC/CustomerDaoImplAbstractTest.java]: You didn't test your implementations CustomerDao

**Pull request merged**